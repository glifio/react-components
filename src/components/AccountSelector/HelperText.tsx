import React from 'react'
import { string } from 'prop-types'
import { Text } from '../Typography'
import { StyledATag } from '../Link'

const Help = () => (
  <>
    <Text mt={2}>
      {"Don't see an account you're looking for? "}
      <StyledATag
        fontSize={2}
        ml={2}
        href='https://reading.supply/@glif/not-seeing-the-right-address-when-accessing-the-glif-wallet-NE1FhV'
      >
        Learn More
      </StyledATag>
    </Text>
  </>
)

const HelperText = ({ text }) => {
  return (
    <>
      <Text>{text}</Text>
      <Help />
    </>
  )
}

HelperText.propTypes = {
  text: string.isRequired
}

export default HelperText
