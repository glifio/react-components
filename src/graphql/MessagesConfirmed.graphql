query MessagesConfirmed($address: String!, $limit: Int!, $offset: Int!) {
  messagesConfirmed(address: $address, limit: $limit, offset: $offset) {
    cid
    height
    from {
      id
      robust
    }
    to {
      id
      robust
    }
    value
    gasFeeCap
    gasPremium
    gasLimit
    methodName
    method
    actorFamily
    exitCode
    gasUsed
    parentBaseFee
    baseFeeBurn
    overEstimationBurn
    minerPenalty
    minerTip
    refund
    gasRefund
    gasBurned
    block {
      Cid
      Timestamp
    }
    params
  }
}
