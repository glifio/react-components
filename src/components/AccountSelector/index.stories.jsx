import { ApolloProvider } from '@apollo/client'
import { CoinType } from '@glif/filecoin-address'
import Filecoin, { HDWalletProvider } from '@glif/filecoin-wallet-provider'
import WalletProviderWrapper, {
  initialLedgerState,
  initialMetaMaskState
} from '../../services/WalletProvider'
import { client } from '../HistoryTables/apolloClient'

import AccountSelector from './index'
import { LoginOption } from '../../customPropTypes'
import { createPath } from '../../utils'
import { FilecoinNumber } from '@glif/filecoin-number'

const LOTUS_API_ADDR = 'https://api.calibration.node.glif.io'

const SEED =
  'soda wage derive hard share hybrid drum company link size crush track inherit entire filter promote resource vocal cycle kind day intact palace mad'

const initialState = {
  loginOption: LoginOption.IMPORT_MNEMONIC,
  walletProvider: new Filecoin(new HDWalletProvider(SEED), {
    apiAddress: LOTUS_API_ADDR
  }),
  error: '',
  ledger: initialLedgerState,
  metamask: initialMetaMaskState,
  wallets: [
    {
      path: createPath(1, 0),
      balance: new FilecoinNumber('0', 'fil'),
      address: 't1mjvkd3nbiarnupgrw23anvft4zjwe4v54v3o44y',
      id: '',
      robust: 't1mjvkd3nbiarnupgrw23anvft4zjwe4v54v3o44y'
    }
  ],
  selectedWalletIdx: 0
}

export default {
  title: 'AccountSelector/AccountSelector',
  component: AccountSelector,
  decorators: [
    Story => (
      <ApolloProvider client={client}>
        <WalletProviderWrapper
          initialState={initialState}
          coinType={CoinType.TEST}
          lotusApiAddr='https://api.calibration.node.glif.io'
        >
          <div style={{ display: 'flex', justifyContent: 'center' }}>
            {Story()}
          </div>
        </WalletProviderWrapper>
      </ApolloProvider>
    )
  ]
}

const Template = args => <AccountSelector {...args} />

export const Base = Template.bind({})
Base.args = {
  onSelectAccount: () => {},
  showSelectedAccount: true,
  helperText: 'Select an account',
  title: 'Account Selector',
  coinType: CoinType.TEST,
  nWalletsToLoad: 5
}
