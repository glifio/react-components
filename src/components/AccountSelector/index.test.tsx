import React from 'react'
import { render, act, screen, waitFor } from '@testing-library/react'
import { CoinType } from '@glif/filecoin-address'

import AccountSelector from '.'

import composeMockAppTree from '../../test-utils/composeMockAppTree'
import { MockedProvider } from '@apollo/client/testing'
import { AddressDocument } from '../../generated/graphql'
import {
  WALLET_ADDRESS,
  WALLET_ADDRESS_2,
  WALLET_ADDRESS_3,
  WALLET_ID,
  WALLET_ID_2,
  WALLET_ID_3
} from '../../test-utils/constants'

jest
  .spyOn(require('./loadNextAccount'), 'loadNextAccount')
  .mockImplementation(
    (
      index,
      provider,
      coinType,
      getBalance,
      getAddress,
      setLoadedFirstWallet,
      setLoadingPage,
      setLoadingWallets,
      _walletList
    ) => {
      // @ts-ignore
      setLoadingWallets(false)
      return Promise.resolve()
    }
  )

jest.mock('../../services/WalletProvider')

const addrs = [
  { robust: WALLET_ADDRESS, id: WALLET_ID },
  { robust: WALLET_ADDRESS_2, id: WALLET_ID_2 },
  { robust: WALLET_ADDRESS_3, id: WALLET_ID_3 }
]

const apolloMocks = addrs.map(({ robust, id }) => {
  return {
    request: {
      query: AddressDocument,
      variables: {
        address: robust
      }
    },
    result: {
      data: {
        address: {
          robust,
          id
        }
      }
    }
  }
})

describe('AccountSelector', () => {
  test('it displays the loading screen first', async () => {
    const { Tree } = composeMockAppTree('multiAccount')
    await act(async () => {
      let res = render(
        <MockedProvider mocks={apolloMocks}>
          <Tree>
            <AccountSelector
              title=''
              helperText=''
              coinType={CoinType.TEST}
              onSelectAccount={() => {}}
            />
          </Tree>
        </MockedProvider>
      )
      expect(screen.getByText(/Loading/)).toBeInTheDocument()
      expect(res.container).toMatchSnapshot()
    })
  })

  test('it displays the accounts in a table view and option to load next account', async () => {
    const { Tree } = composeMockAppTree('multiAccount')
    await act(async () => {
      let res = render(
        <MockedProvider mocks={apolloMocks}>
          <Tree>
            <AccountSelector
              title='Select Account'
              helperText=''
              coinType={CoinType.TEST}
              onSelectAccount={() => {}}
            />
          </Tree>
        </MockedProvider>
      )

      await waitFor(() => {
        expect(screen.getByText(/Select Account/)).toBeInTheDocument()
        addrs.forEach(({ robust, id }, i) => {
          expect(screen.getByText(robust)).toBeInTheDocument()
          expect(screen.getByText(id)).toBeInTheDocument()
          const role = screen.getAllByRole('link')[i]
          expect(
            role
              .getAttribute('href')
              ?.includes(`/actor/?address=${robust || id}`)
          )
        })

        expect(screen.getByText(/Next account/)).toBeInTheDocument()

        expect(res.container).toMatchSnapshot()
      })
    })
  })

  test('it displays the selected account when showSelectedACcount is passed', async () => {
    const { Tree } = composeMockAppTree('multiAccount')
    await act(async () => {
      let res = render(
        <MockedProvider mocks={apolloMocks}>
          <Tree>
            <AccountSelector
              title='Select Account'
              helperText=''
              coinType={CoinType.TEST}
              onSelectAccount={() => {}}
              showSelectedAccount
            />
          </Tree>
        </MockedProvider>
      )

      await waitFor(() => {
        expect(screen.getByTestId('selected-account')).toBeInTheDocument()
        expect(res.container).toMatchSnapshot()
      })
    })
  })
})
