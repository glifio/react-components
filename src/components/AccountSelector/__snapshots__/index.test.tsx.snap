// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSelector it displays the accounts in a table view and option to load next account 1`] = `
.c8 {
  width: 24;
  height: 24;
}

.c0 {
  padding: 1.5em;
  border-radius: 8px;
  overflow: hidden;
  text-align: center;
  word-break: break-word;
}

.c0 > *:first-child {
  margin-top: 0;
}

.c0 > *:last-child {
  margin-bottom: 0;
}

.c0 > svg:first-child {
  display: block;
  margin: 0 auto 1em;
}

.c0 > h2:first-child {
  margin: 0;
}

.c0 header:first-child {
  margin: -1.5em -1.5em 1.5em -1.5em;
  padding: 1.5em;
  background-color: var(--purple-medium);
  color: var(--white);
}

.c1 {
  background-color: var(--blue-gray);
}

.c1 hr {
  border-color: var(--gray-medium);
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: var(--space-m);
  margin: 1.5em 0;
}

.c14 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  min-width: 0;
  box-sizing: border-box;
  color: var(--purple-medium);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  height: auto;
  cursor: pointer;
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  background: transparent;
  padding: 0;
  outline: none;
}

.c10 {
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
}

.c6:hover .c9 {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
}

.c12 {
  -webkit-transition: 0.18s ease-in;
  transition: 0.18s ease-in;
  opacity: 0;
  text-align: left;
  color: var(--purple-medium);
  margin-top: 0;
  margin-left: 4px;
  margin-top: 0;
  margin-left: 4px;
}

.c6:hover .c11 {
  opacity: 1;
}

.c17 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  gap: 0.75rem;
  padding: 0.5em 0.75em;
  color: var(--black);
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  border-color: var(--black);
  background: transparent;
  font-size: 1.25em;
}

.c17 svg path {
  stroke: var(--black);
}

.c17:hover {
  color: var(--purple-medium);
  border-color: var(--purple-medium);
}

.c17:hover svg path {
  stroke: var(--purple-medium);
}

.c17:active {
  color: var(--white) !important;
  border-color: var(--purple-medium) !important;
  background: var(--purple-medium) !important;
}

.c17:active svg path {
  stroke: var(--white) !important;
}

.c2 {
  position: relative;
  display: inline-block;
  width: 18px;
  height: 18px;
  border-width: 1px;
  border-style: solid;
  border-radius: 50%;
  margin: 4px 0 0 1em;
  border-color: var(--purple-medium);
}

.c2:after {
  position: absolute;
  display: block;
  content: '';
  top: 50%;
  left: 50%;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c2:after {
  background-color: var(--purple-medium);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-gap: 0.3em;
  line-height: 1.5;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  line-height: 1;
}

.c4 > svg {
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
}

.c4 > svg:hover {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
  fill: var(--purple-medium);
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: var(--space-m);
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c15 > *:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c15 > *:last-child:not(.toggle-wrapper) {
  -webkit-flex: 0 0 50%;
  -ms-flex: 0 0 50%;
  flex: 0 0 50%;
}

.c15 span.info,
.c15 span.error {
  font-size: 0.875rem;
}

.c15 span.error {
  color: var(--red-medium);
}

.c16 .toggle-wrapper {
  position: relative;
}

.c16 input {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 0;
  height: 0;
  opacity: 0;
}

.c16 .toggle {
  display: inline-block;
  position: relative;
  width: 2.25em;
  height: 1.5em;
  border-radius: 0.75em;
  border: 2px solid var(--blue-medium);
  background-color: var(--blue-medium);
}

.c16 .toggle::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: calc(1.5em - 4px);
  height: calc(1.5em - 4px);
  border-radius: 50%;
  background-color: var(--white);
}

.c16 input:checked ~ .toggle {
  border-color: var(--blue-dark);
  background-color: var(--blue-dark);
}

.c16 input:checked ~ .toggle::after {
  left: auto;
  right: 0;
}

<div>
  <div>
    <div
      class="c0 c1"
    >
      <h2>
        Select Account
      </h2>
      <p />
    </div>
    <br />
    <table>
      <thead>
        <tr>
          <th />
          <th>
            #
          </th>
          <th>
            Name
          </th>
          <th>
            Address
          </th>
          <th>
            Balance
          </th>
          <th>
            Id
          </th>
          <th>
            Path
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="selectable"
        >
          <td
            data-testid="selected-account"
          >
            <div
              class="c2"
              color="purple"
            />
          </td>
          <td>
            0
          </td>
          <td>
            Account 
            0
          </td>
          <td>
            <div
              class="c3"
            >
              t1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi
              <a
                class="c4"
                href="undefined/actor/?address=t1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  height="18px"
                  viewBox="0 0 24 24"
                  width="18px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19,21H5c-1.1,0-2-0.9-2-2V5c0-1.1,0.9-2,2-2h7v2H5v14h14v-7h2v7C21,20.1,20.1,21,19,21z"
                  />
                  <path
                    d="M21 10L19 10 19 5 14 5 14 3 21 3z"
                  />
                  <path
                    d="M6.7 8.5H22.3V10.5H6.7z"
                    transform="rotate(-45.001 14.5 9.5)"
                  />
                </svg>
              </a>
              <div
                class="c5"
                color="var(--purple-medium)"
                display="flex"
              >
                <button
                  class="c6 c7"
                  display="flex"
                  role="button"
                  type="button"
                >
                  <svg
                    class="c8 c9 c10"
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="#1A0066"
                      height="12"
                      width="12"
                      x="8"
                      y="4"
                    />
                    <rect
                      fill="#E0D7FE"
                      height="12"
                      width="12"
                      x="4"
                      y="8"
                    />
                  </svg>
                  <span
                    class="c11 c12"
                    color="var(--purple-medium)"
                  >
                    Copy
                  </span>
                </button>
              </div>
            </div>
          </td>
          <td>
            1
          </td>
          <td>
            t0101
          </td>
          <td>
            m/44'/1'/0'/0/0
          </td>
        </tr>
        <tr
          class="selectable"
        >
          <td />
          <td>
            1
          </td>
          <td>
            Account 
            1
          </td>
          <td>
            <div
              class="c3"
            >
              t1nq5k2mps5umtebdovlyo7y6a3ywc7u4tobtuo3a
              <a
                class="c4"
                href="undefined/actor/?address=t1nq5k2mps5umtebdovlyo7y6a3ywc7u4tobtuo3a"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  height="18px"
                  viewBox="0 0 24 24"
                  width="18px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19,21H5c-1.1,0-2-0.9-2-2V5c0-1.1,0.9-2,2-2h7v2H5v14h14v-7h2v7C21,20.1,20.1,21,19,21z"
                  />
                  <path
                    d="M21 10L19 10 19 5 14 5 14 3 21 3z"
                  />
                  <path
                    d="M6.7 8.5H22.3V10.5H6.7z"
                    transform="rotate(-45.001 14.5 9.5)"
                  />
                </svg>
              </a>
              <div
                class="c5"
                color="var(--purple-medium)"
                display="flex"
              >
                <button
                  class="c6 c7"
                  display="flex"
                  role="button"
                  type="button"
                >
                  <svg
                    class="c8 c9 c10"
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="#1A0066"
                      height="12"
                      width="12"
                      x="8"
                      y="4"
                    />
                    <rect
                      fill="#E0D7FE"
                      height="12"
                      width="12"
                      x="4"
                      y="8"
                    />
                  </svg>
                  <span
                    class="c11 c12"
                    color="var(--purple-medium)"
                  >
                    Copy
                  </span>
                </button>
              </div>
            </div>
          </td>
          <td>
            5
          </td>
          <td>
            t0102
          </td>
          <td>
            m/44'/1'/0'/0/1
          </td>
        </tr>
        <tr
          class="selectable"
        >
          <td />
          <td>
            2
          </td>
          <td>
            Account 
            2
          </td>
          <td>
            <div
              class="c3"
            >
              t1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq
              <a
                class="c4"
                href="undefined/actor/?address=t1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  height="18px"
                  viewBox="0 0 24 24"
                  width="18px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19,21H5c-1.1,0-2-0.9-2-2V5c0-1.1,0.9-2,2-2h7v2H5v14h14v-7h2v7C21,20.1,20.1,21,19,21z"
                  />
                  <path
                    d="M21 10L19 10 19 5 14 5 14 3 21 3z"
                  />
                  <path
                    d="M6.7 8.5H22.3V10.5H6.7z"
                    transform="rotate(-45.001 14.5 9.5)"
                  />
                </svg>
              </a>
              <div
                class="c5"
                color="var(--purple-medium)"
                display="flex"
              >
                <button
                  class="c6 c7"
                  display="flex"
                  role="button"
                  type="button"
                >
                  <svg
                    class="c8 c9 c10"
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="#1A0066"
                      height="12"
                      width="12"
                      x="8"
                      y="4"
                    />
                    <rect
                      fill="#E0D7FE"
                      height="12"
                      width="12"
                      x="4"
                      y="8"
                    />
                  </svg>
                  <span
                    class="c11 c12"
                    color="var(--purple-medium)"
                  >
                    Copy
                  </span>
                </button>
              </div>
            </div>
          </td>
          <td>
            0
          </td>
          <td>
            t0103
          </td>
          <td>
            m/44'/1'/0'/0/2
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="c13 c14"
    >
      <label
        class="c15 c16"
      >
        <div>
          <span>
            Expert Mode
          </span>
          
        </div>
        <div
          class="toggle-wrapper"
        >
          <input
            type="checkbox"
          />
          <span
            class="toggle"
          />
        </div>
      </label>
      <button
        class="c17"
      >
        Next account
      </button>
    </div>
  </div>
</div>
`;

exports[`AccountSelector it displays the loading screen first 1`] = `
.c1 {
  position: relative;
  display: inline-block;
  background-color: var(--purple-medium);
  width: 3em;
  height: 3em;
  margin: 0;
}

.c1:after {
  position: absolute;
  display: block;
  content: '';
  top: 0;
  left: 0;
  background-color: var(--purple-dark);
  -webkit-animation-name: gxoozo;
  animation-name: gxoozo;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  width: calc(3em * 0.5);
  height: calc(3em * 0.5);
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    />
    <p>
      Loading...
    </p>
  </div>
</div>
`;

exports[`AccountSelector it displays the selected account when showSelectedACcount is passed 1`] = `
.c3 {
  position: relative;
  display: inline-block;
  background-color: var(--purple-medium);
  width: 2em;
  height: 2em;
  margin: 0;
}

.c3:after {
  position: absolute;
  display: block;
  content: '';
  top: 0;
  left: 0;
  background-color: var(--purple-dark);
  -webkit-animation-name: gxoozo;
  animation-name: gxoozo;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  width: calc(2em * 0.5);
  height: calc(2em * 0.5);
}

.c10 {
  width: 24;
  height: 24;
}

.c0 {
  padding: 1.5em;
  border-radius: 8px;
  overflow: hidden;
  text-align: center;
  word-break: break-word;
}

.c0 > *:first-child {
  margin-top: 0;
}

.c0 > *:last-child {
  margin-bottom: 0;
}

.c0 > svg:first-child {
  display: block;
  margin: 0 auto 1em;
}

.c0 > h2:first-child {
  margin: 0;
}

.c0 header:first-child {
  margin: -1.5em -1.5em 1.5em -1.5em;
  padding: 1.5em;
  background-color: var(--purple-medium);
  color: var(--white);
}

.c1 {
  background-color: var(--blue-gray);
}

.c1 hr {
  border-color: var(--gray-medium);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: var(--space-l);
}

.c7 {
  min-width: 0;
  box-sizing: border-box;
  color: var(--purple-medium);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  height: auto;
  cursor: pointer;
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  background: transparent;
  padding: 0;
  outline: none;
}

.c12 {
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
}

.c8:hover .c11 {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
}

.c14 {
  -webkit-transition: 0.18s ease-in;
  transition: 0.18s ease-in;
  opacity: 0;
  text-align: left;
  color: var(--purple-medium);
  margin-top: 0;
  margin-left: 4px;
  margin-top: 0;
  margin-left: 4px;
}

.c8:hover .c13 {
  opacity: 1;
}

.c4 {
  position: relative;
  display: inline-block;
  width: 18px;
  height: 18px;
  border-width: 1px;
  border-style: solid;
  border-radius: 50%;
  margin: 4px 0 0 1em;
  border-color: var(--purple-medium);
}

.c4:after {
  position: absolute;
  display: block;
  content: '';
  top: 50%;
  left: 50%;
  width: 11px;
  height: 11px;
  border-radius: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c4:after {
  background-color: var(--purple-medium);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-gap: 0.3em;
  line-height: 1.5;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  line-height: 1;
}

.c6 > svg {
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
}

.c6 > svg:hover {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
  fill: var(--purple-medium);
}

<div>
  <div>
    <div
      class="c0 c1"
    >
      <h2>
        Select Account
      </h2>
      <p />
    </div>
    <br />
    <table>
      <caption>
        <div
          class="c2"
        >
          <span
            class="c3"
          />
          <span>
            Loading ...
          </span>
        </div>
      </caption>
      <thead>
        <tr>
          <th />
          <th>
            #
          </th>
          <th>
            Name
          </th>
          <th>
            Address
          </th>
          <th>
            Balance
          </th>
          <th>
            Id
          </th>
          <th>
            Path
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="selectable"
        >
          <td
            data-testid="selected-account"
          >
            <div
              class="c4"
              color="purple"
            />
          </td>
          <td>
            0
          </td>
          <td>
            Account 
            0
          </td>
          <td>
            <div
              class="c5"
            >
              t1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi
              <a
                class="c6"
                href="undefined/actor/?address=t1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  height="18px"
                  viewBox="0 0 24 24"
                  width="18px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19,21H5c-1.1,0-2-0.9-2-2V5c0-1.1,0.9-2,2-2h7v2H5v14h14v-7h2v7C21,20.1,20.1,21,19,21z"
                  />
                  <path
                    d="M21 10L19 10 19 5 14 5 14 3 21 3z"
                  />
                  <path
                    d="M6.7 8.5H22.3V10.5H6.7z"
                    transform="rotate(-45.001 14.5 9.5)"
                  />
                </svg>
              </a>
              <div
                class="c7"
                color="var(--purple-medium)"
                display="flex"
              >
                <button
                  class="c8 c9"
                  display="flex"
                  role="button"
                  type="button"
                >
                  <svg
                    class="c10 c11 c12"
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="#1A0066"
                      height="12"
                      width="12"
                      x="8"
                      y="4"
                    />
                    <rect
                      fill="#E0D7FE"
                      height="12"
                      width="12"
                      x="4"
                      y="8"
                    />
                  </svg>
                  <span
                    class="c13 c14"
                    color="var(--purple-medium)"
                  >
                    Copy
                  </span>
                </button>
              </div>
            </div>
          </td>
          <td>
            1
          </td>
          <td>
            t0101
          </td>
          <td>
            m/44'/1'/0'/0/0
          </td>
        </tr>
        <tr
          class="selectable"
        >
          <td />
          <td>
            1
          </td>
          <td>
            Account 
            1
          </td>
          <td>
            <div
              class="c5"
            >
              t1nq5k2mps5umtebdovlyo7y6a3ywc7u4tobtuo3a
              <a
                class="c6"
                href="undefined/actor/?address=t1nq5k2mps5umtebdovlyo7y6a3ywc7u4tobtuo3a"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  height="18px"
                  viewBox="0 0 24 24"
                  width="18px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19,21H5c-1.1,0-2-0.9-2-2V5c0-1.1,0.9-2,2-2h7v2H5v14h14v-7h2v7C21,20.1,20.1,21,19,21z"
                  />
                  <path
                    d="M21 10L19 10 19 5 14 5 14 3 21 3z"
                  />
                  <path
                    d="M6.7 8.5H22.3V10.5H6.7z"
                    transform="rotate(-45.001 14.5 9.5)"
                  />
                </svg>
              </a>
              <div
                class="c7"
                color="var(--purple-medium)"
                display="flex"
              >
                <button
                  class="c8 c9"
                  display="flex"
                  role="button"
                  type="button"
                >
                  <svg
                    class="c10 c11 c12"
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="#1A0066"
                      height="12"
                      width="12"
                      x="8"
                      y="4"
                    />
                    <rect
                      fill="#E0D7FE"
                      height="12"
                      width="12"
                      x="4"
                      y="8"
                    />
                  </svg>
                  <span
                    class="c13 c14"
                    color="var(--purple-medium)"
                  >
                    Copy
                  </span>
                </button>
              </div>
            </div>
          </td>
          <td>
            5
          </td>
          <td>
            t0102
          </td>
          <td>
            m/44'/1'/0'/0/1
          </td>
        </tr>
        <tr
          class="selectable"
        >
          <td />
          <td>
            2
          </td>
          <td>
            Account 
            2
          </td>
          <td>
            <div
              class="c5"
            >
              t1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq
              <a
                class="c6"
                href="undefined/actor/?address=t1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  height="18px"
                  viewBox="0 0 24 24"
                  width="18px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19,21H5c-1.1,0-2-0.9-2-2V5c0-1.1,0.9-2,2-2h7v2H5v14h14v-7h2v7C21,20.1,20.1,21,19,21z"
                  />
                  <path
                    d="M21 10L19 10 19 5 14 5 14 3 21 3z"
                  />
                  <path
                    d="M6.7 8.5H22.3V10.5H6.7z"
                    transform="rotate(-45.001 14.5 9.5)"
                  />
                </svg>
              </a>
              <div
                class="c7"
                color="var(--purple-medium)"
                display="flex"
              >
                <button
                  class="c8 c9"
                  display="flex"
                  role="button"
                  type="button"
                >
                  <svg
                    class="c10 c11 c12"
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="#1A0066"
                      height="12"
                      width="12"
                      x="8"
                      y="4"
                    />
                    <rect
                      fill="#E0D7FE"
                      height="12"
                      width="12"
                      x="4"
                      y="8"
                    />
                  </svg>
                  <span
                    class="c13 c14"
                    color="var(--purple-medium)"
                  >
                    Copy
                  </span>
                </button>
              </div>
            </div>
          </td>
          <td>
            0
          </td>
          <td>
            t0103
          </td>
          <td>
            m/44'/1'/0'/0/2
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
